### DISCUSSION TITLE
`state does not match`

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#30101] [Other Feature Feedback, Questions, & Ideas] [answered]
My rebooted for an OS upgrade and I reopened Chrome which restored my tabs.

One of the tabs says:

https://cs.github.com/auth/callback?code=GHSAT0AAAAAABQR2YQAUT75U6EIMADK6NO4YX7RG7A&state=3c74f079-4af4-4348-8e69-6495a11a96d2

> state does not match

Loading that url in incognito I get:

> failed to get nonce from session

I'm sure these are true statements. They also are not a good UX.

Disclaimer: I'm aware that these are potentially secret things, given how void their values are, I'm consciously posting them. If it turns out they're dangerous, that's another bug that GitHub should also resolve. (At the very least, the error message should say "please don't repost this URL beyond sharing it to support" or something to that effect.)

## Expected results

Load the app (or the teaser app) and report the message inside that. Don't give humans plaintext errors.

---

<details><summary>
I should note that if I navigate "back" in this tab, I get back to a working page with useful content. Obviously my preference for when my browser restarts is that each page has useful content.
</summary>

For the curious, my actual url was something akin to:
https://cs.github.com/?q=abcdefghijklmopqrstuvwxyz&p=2

<img width="822" alt="image" src="https://user-images.githubusercontent.com/2119212/185650128-321828a2-a9d9-4e70-86a9-ad564f42f71d.png">
</details>

**Comment by colinwm** [comment, marked as answer]
Hey @jsoref, I guess your old browser tab was part way through authentication. Those `state` and `code` parameters expire pretty quickly, so they were likely just expired. Sorry for the ugly error message :smile:

**Reply by jsoref** [reply]
Thanks @colinwm, I'm hoping you'll improve the error screen 😉.

(We all start w/ lousy error reporting, but we can pave the cowpaths 🐮 ➡️  or desire lines 🙏 📈. -- I'm currently working on this for my project...)
